### DISCUSSION TITLE
Réflexions sur la gestion du site [Thoughts on the management of the site]

### DISCUSSION THREAD

**Discussion by joaopa00** [Catch-up-TV-and-More/plugin.video.catchuptvandmore#659] [Q&A]
**English-speaker: see below**

Avant toutes choses, je ne fait pas partie des propriétaires du site. Je ne suis qu'une petite main qui aide à la gestion (et depuis seulement une semaine :D :D). Ce qui suit n'engage donc que moi et pas les propriétaires du site (wwark et sy6sy2).
Voici quelques réflexions qui je pense pour améliorer l'application:

Catchuptvmore est un logiciel qui dépend de contraintes extéerieures nombreuses (changement d'api des chaines, changement d'URL...). Comme on ne peut les maîtriser, Catchuptvmore perd de son intêret si on ne le met pas à jour fréquemment. Pour éviter ce problème, je propose de faire une mise à jour régulière (disons tous les mois), sauf énorme problème où l'on pourrait réagir encore plus vite.

-Faire une petite gagnotte pour récolter quelques piécettes pour payer un vpn. Les sites sont souvent géolocalisés et seules les locaux peuvent développer te résoudre les problèmes rencontrés. Achter un vpn résoudrait ce problème. 

-Utilisation de Flake8. Ce logiciel est conçu pour uniformiser le style d'écriture en python. Cependant certains choix de styles sont discutables (voire contestables). Catchuptvmore doit-il choisir et fixer son propre style.

Qu'en pensez-vous?

Si vous avez d'autres idées ou souhaits, n'hésitez pas à les exprimer.

Joaopa

For our overseas friends:

First of all, I am not one of the owners of the site. I'm just a little hand helping with the management (and only been for one week :D :D). What follows therefore engages only me and not the owners of the site (wwark and sy6sy2).
Here are some thoughts that I think to improve the app:

Catchuptvmore is a software that depends on many external constraints (change of channel API, change of URL...). As we cannot control them, Catchuptvmore loses its interest if we do not update it frequently. To avoid this problem, I propose to do a regular update (say every month), except for a huge problem where we could react even faster.

-Make a little jackpot to collect a few coins to pay for a vpn. The sites are often geolocated and only the locals can develop and solve the problems encountered. Buying a vpn would solve this problem.

-Use of Flake8. This software is designed to standardize the style of writing in python. However, some style choices are debatable (even questionable). Should Catchuptvmore choose and set its own style. 

What do you think?

If you have other ideas or wishes, do not hesitate to express them. 

Joaopa

**Comment by darodi** [comment]
je vois que tous les éditeurs ne gèrent pas de la même façon le pep8 apparemment.

pycharm renvoie un warning pep8 pour les lignes > 120 caractères.
Or, flake8  ne les accepte pas sur plusieurs lignes dans certains cas:

![image](https://user-images.githubusercontent.com/4682830/156464986-42c637af-e960-46f3-8f91-802c5d707adb.png)

```
flake8 tv5mondeplus.py
tv5mondeplus.py:30:29: E126 continuation line over-indented for hanging indent
tv5mondeplus.py:34:34: E126 continuation line over-indented for hanging indent
tv5mondeplus.py:40:19: E126 continuation line over-indented for hanging indent
tv5mondeplus.py:44:39: E126 continuation line over-indented for hanging indent
```


-----------------


n'y aurait-il pas moyen d'intégrer correctement flake8 au CI ?
exemple: https://github.com/Catch-up-TV-and-More/plugin.video.catchuptvandmore/runs/5392103276?check_suite_focus=true
je ne vois pas l'erreur dans le pull request
il y a moyen de corriger ceci?
https://github.com/Catch-up-TV-and-More/plugin.video.catchuptvandmore/actions/workflows/ci.yml

je vois qu'il existe d'autres intégration github de flake8
https://github.com/marketplace/actions/run-flake8-on-your-pr-with-annotations
https://github.com/reviewdog/action-flake8
...

**Reply by darodi** [reply]
voir #686

**Comment by joaopa00** [comment]
Je suis aussi géné par Flake 8. En effet, des erreurs ne sont pas détectées. Si tu connaîs un moyen d'éviter cela, j'essaierai d'intégré tes modifications. Je dois t'avouer que je ne vois ce qu'apporte de plus action-flake8 par rapport à ce que l'on a déjà. Quant à l'autre site, il me semble complètement périmé. Peux-tu développer ton point de vue?

Merci d'avance.

**Reply by darodi** [reply]
voir #686
